mailing_list/models.py

```diff
@@ -1,33 +1,152 @@
+                    'address': mailing_list.address,
+                    'access_level': access_level,
+                    'members_count': members_count,
+                    'is_primary_section': s['sis_section_id'] is not None
+                }
+
+            cache.set(cache_key, lists)
+
+        return lists.values()
 
 
 class MailingList(models.Model):
     canvas_course_id = models.IntegerField()
-    section_id = models.IntegerField(null=True, blank=True)
-    active = models.BooleanField(default=False)
+    section_id = models.IntegerField()
     created_by = models.CharField(max_length=32)
     modified_by = models.CharField(max_length=32)
     date_created = models.DateTimeField(blank=True, default=datetime.utcnow)
     date_modified = models.DateTimeField(blank=True, default=datetime.utcnow)
     subscriptions_updated = models.DateTimeField(null=True, blank=True)
 
+    objects = MailingListManager()
+
     class Meta:
         db_table = 'ml_mailing_list'
         unique_together = ('canvas_course_id', 'section_id')
 
     def __unicode__(self):
-        return u'canvas_course_id: {}, section_id: {}, active: {}'.format(
+        return u'canvas_course_id: {}, section_id: {}'.format(
             self.canvas_course_id,
-            self.section_id,
-            self.active
+            self.section_id
         )
 
     @property
     def address(self):
-        return "canvas-%s@%s" % (self.canvas_course_id, settings.LISTSERV_DOMAIN)
+        return "canvas-%s-%s@%s" % (self.canvas_course_id, self.section_id, settings.LISTSERV_DOMAIN)
+
+    def update_access_level(self, access_level):
+        logger.debug("Updating access_level for listserv mailing list %s with %s", self.address, access_level)
+        listserv_client.update_list(self, access_level)
+        cache.delete(settings.CACHE_KEY_LISTS_BY_CANVAS_COURSE_ID % self.canvas_course_id)
+        logger.debug("Finished updating listserv mailing list for canvas_course_id %s", self.canvas_course_id)
+
+    def sync_listserv_membership(self):
+        logger.debug("Synchronizing listserv membership for canvas course id %s", self.canvas_course_id)
+
+        canvas_course_id = self.canvas_course_id
+        unsubscribed = {x.email for x in self.unsubscribed_set.all()}
+
+        try:
+            canvas_enrollments = get_all_list_data(SDK_CONTEXT, enrollments.list_enrollments_courses, canvas_course_id)
+        except CanvasAPIError:
+            logger.exception("Failed to get canvas enrollments for canvas_course_id %s", canvas_course_id)
+            raise
+
+        univ_ids = []
+        for enrollment in canvas_enrollments:
+            try:
+                univ_ids.append(enrollment['user']['sis_user_id'])
+            except KeyError:
+                logger.debug("Found canvas enrollment with missing sis_user_id %s", json.dumps(enrollment, indent=4))
+        enrolled_emails = set([p.email_address for p in Person.objects.filter(univ_id__in=univ_ids)])
```

Comment:
nitpick: set comprehensions are more readable, imo

(btw, in case it's not clear, feel free to ignore all my nitpicks)